internal/repository/attestations.go

```diff
@@ -255,6 +267,188 @@ func (r *Repository) AddGitHubPullRequestAttestationForNumber(ctx context.Contex
 	return r.addGitHubPullRequestAttestation(ctx, signer, owner, repository, pullRequest, signCommit)
 }
 
+// AddGitHubPullRequestApprover adds a GitHub pull request approval attestation
+// for the specified parameters. If an attestation already exists, the specified
+// approver is added to the existing attestation's predicate and it is re-signed
+// and stored in the repository. Currently, this is limited to developer mode.
+func (r *Repository) AddGitHubPullRequestApprover(ctx context.Context, signer sslibdsse.SignerVerifier, owner, repository string, pullRequestNumber int, reviewID int64, approver *tuf.Key, signCommit bool) error {
+	if !dev.InDevMode() {
+		return dev.ErrNotInDevMode
+	}
+
+	currentAttestations, err := attestations.LoadCurrentAttestations(r.r)
+	if err != nil {
+		return err
+	}
+
+	baseRef, fromID, toID, err := getGitHubPullRequestReviewDetails(ctx, currentAttestations, owner, repository, pullRequestNumber, reviewID)
+	if err != nil {
+		return err
+	}
+
+	// TODO: if the helper above has an indexPath, we can directly load that blob, simplifying the logic here
+	hasApprovalAttestation := false
+	env, err := currentAttestations.GetGitHubPullRequestApprovalAttestationFor(r.r, baseRef, fromID, toID)
+	if err == nil {
+		slog.Debug("Found existing GitHub pull request approval attestation...")
+		hasApprovalAttestation = true
+	} else if !errors.Is(err, attestations.ErrGitHubPullRequestApprovalAttestationNotFound) {
+		return err
+	}
+
+	approvers := []*tuf.Key{approver}
+	var dismissedApprovers []*tuf.Key
+	if !hasApprovalAttestation {
+		// Create a new GitHub pull request approval attestation
+		slog.Debug("Creating new GitHub pull request approval attestation...")
+	} else {
+		// Update existing statement's predicate and create new env
+		slog.Debug("Adding approver to existing GitHub pull request approval attestation...")
+		payloadBytes, err := env.DecodeB64Payload()
+		if err != nil {
+			return err
+		}
+
+		type tmpStatement struct {
+			Type          string                                             `json:"_type"`
+			Subject       []*ita.ResourceDescriptor                          `json:"subject"`
+			PredicateType string                                             `json:"predicateType"`
+			Predicate     *attestations.GitHubPullRequestApprovalAttestation `json:"predicate"`
+		}
+		stmt := &tmpStatement{Subject: []*ita.ResourceDescriptor{}, Predicate: &attestations.GitHubPullRequestApprovalAttestation{Approvers: []*tuf.Key{}, DismissedApprovers: []*tuf.Key{}}}
```

Comment:
Rule of thumb: only one-line if it's a very simple struct (usually 1 field)
```suggestion
		stmt := &tmpStatement{
		  Subject: []*ita.ResourceDescriptor{}, 
		  Predicate: &attestations.GitHubPullRequestApprovalAttestation{
		    Approvers: []*tuf.Key{},
		    DismissedApprovers: []*tuf.Key{},
		  }
		}
```

Reply:
gotcha, will fix.